src/presentation.rs

```diff
@@ -292,6 +308,22 @@ impl PresentationState {
 
     fn set_current_slide_index(&self, value: usize) {
         self.inner.deref().borrow_mut().current_slide_index = value;
+        if let Some(SpeakerNoteChannel::Notifier(notifier)) = &self.inner.deref().borrow().channel {
```

Comment:
Following the idea of moving this to presenter, you can always emit this event after a command is handled. That would help detach this logic from out of here.